### DISCUSSION TITLE
Support to ESP8266 using ENC28J60

### DISCUSSION THREAD

**Discussion by khoih-prog** [khoih-prog/AsyncHTTPRequest_Generic#29] [Ideas]
Hi @jandrassy

I tried to use ESP8266 and ENC28J60 with [**lwIP_enc28j60**](https://github.com/esp8266/Arduino/tree/master/libraries/lwIP_enc28j60) library without success, even with the core sample example [**TCPClient**](https://github.com/esp8266/Arduino/tree/master/libraries/lwIP_w5500/examples/TCPClient)

I believe you, as the author of [**EthernetENC** library](https://github.com/jandrassy/EthernetENC) have better knowledge in solving the issue / bug, if exists in that  [**lwIP_enc28j60**](https://github.com/esp8266/Arduino/tree/master/libraries/lwIP_enc28j60) library

I'm trying to add Async support to ENC28J60 as I've just done with W5x00, in [AsyncHTTPRequest_Generic releases v1.8.0](https://github.com/khoih-prog/AsyncHTTPRequest_Generic/releases/tag/v1.8.0)

Thanks,

**Comment by khoih-prog** [comment]
Oh sorry, just a bad ENC28J60 board. It's working now with the example [**TCPClient**](https://github.com/esp8266/Arduino/tree/master/libraries/lwIP_w5500/examples/TCPClient)

**Comment by khoih-prog** [comment]
And this [AsyncHTTPRequest_Generic](https://github.com/khoih-prog/AsyncHTTPRequest_Generic) library

```
Starting AsyncHTTPRequest_ESP8266_Ethernet on ESP8266_NODEMCU_ESP12E using ESP8266_ENC28J60 Ethernet
AsyncHTTPRequest_Generic v1.8.0
Connecting to network : ..........................................................
Ethernet IP address: 192.168.2.187

**************************************
abbreviation: EDT
client_ip: aaa.bbb.ccc.ddd
datetime: 2022-04-13T23:54:42.973735-04:00
day_of_week: 3
day_of_year: 103
dst: true
dst_from: 2022-03-13T07:00:00+00:00
dst_offset: 3600
dst_until: 2022-11-06T06:00:00+00:00
raw_offset: -18000
timezone: America/Toronto
unixtime: 1649908482
utc_datetime: 2022-04-14T03:54:42.973735+00:00
utc_offset: -04:00
week_number: 15
**************************************

```

**Comment by JAndrassy** [comment]
I tested the lwIP_enc28j60 when it was new.
there is still one issue https://github.com/esp8266/Arduino/issues/8144

**Comment by khoih-prog** [comment]
Thanks for the info. Will keep in mind.

**Comment by khoih-prog** [comment]
Already implementing / testing static IP OK in 

https://github.com/khoih-prog/AsyncMQTT_Generic/blob/main/examples/ESP8266/FullyFeatured_ESP8266_Ethernet/FullyFeatured_ESP8266_Ethernet.ino#L46-L50

as

```
#if !USING_DHCP
  eth.config(localIP, gateway, netMask, gateway);
#endif
  
  eth.setDefault();
```

**Reply by JAndrassy** [reply]
problem is the setDefault(). it should work without it.
